src/app/rewards/page.tsx

```diff
@@ -0,0 +1,326 @@
+import { DexterToast } from "../components/DexterToaster";
+
+export default function Rewards() {
+  useEffect(() => {
+    initializeSubscriptions(store);
+    return () => {
+      unsubscribeAll();
+    };
+  }, []);
+
+  const { showSuccessUi } = useAppSelector((state) => state.rewardSlice);
+  return (
+    <div className="bg-[#141414]">
+      <HeaderComponent />
+      {showSuccessUi ? <SuccessUi /> : <RewardsCard />}
+    </div>
+  );
+}
+
+function SuccessUi() {
+  const t = useTranslations();
+  const dispatch = useAppDispatch();
+  return (
+    <div className="max-w-[400px] sm:max-w-[600px] rounded-2xl w-full px-4 py-4 sm:px-12 sm:py-8 m-auto mt-2 sm:mt-14 mb-28 bg-[#191B1D]">
+      <div className="flex flex-col items-center pb-6">
+        {/* Title */}
+        <h4 className="font-bold text-center text-3xl !m-0 !my-5">
+          {t("rewards_claimed")} 🎉
+        </h4>
+
+        {/* Subtitle */}
+        <p className="text-base text-center px-2">
+          {t("continue_trading_to_earn_more")}
+        </p>
+
+        {/* "Go Back" button */}
+        <button
+          className={`max-w-[220px] min-h-[44px] uppercase w-full px-4 my-6 mt-8 rounded bg-dexter-green text-black opacity-100`}
+          onClick={() => dispatch(rewardSlice.actions.resetRewardState())}
+        >
+          <span className="font-bold text-sm tracking-[.1px] ">
+            {t("go_back")}
+          </span>
+        </button>
+
+        {/* Send feedback secondary action */}
+        <SecondaryAction
+          textIdentifier="how_did_everything_go"
+          targetUrl="https://t.me/dexter_discussion"
+        />
+      </div>
+    </div>
+  );
+}
+
+function HeaderComponent() {
+  const t = useTranslations();
+  return (
+    <div className="max-w-[400px] sm:max-w-[600px] px-5 sm:p-0 m-auto">
+      <div className="sm:flex sm:justify-center sm:items-center mt-10">
+        <div className="text-white sm:w-[62%]">
+          <div className="flex flex-col justify-center h-full">
+            <DexterHeading title={t("rewards")} />
+            <DexterParagraph text={t("earn_rewards_by")} />
+          </div>
+        </div>
+        <div className="sm:w-[38%] max-[640px]:max-w-[200px] sm:ml-5">
+          <img
+            src="/rewards/chest.png"
+            alt="treasury"
+            className="w-full"
+            width={218}
+          />
+        </div>
+      </div>
+    </div>
+  );
+}
+
+function RewardsCard() {
+  const dispatch = useAppDispatch();
+  const { isConnected, walletData } = useAppSelector((state) => state.radix);
+  const account = walletData.accounts[0]?.address;
+  const t = useTranslations();
+  const { rewardData, pairsList } = useAppSelector(
+    (state) => state.rewardSlice
+  );
+  const userHasRewards = getUserHasRewards(rewardData);
+
+  useEffect(() => {
+    async function loadRewards() {
+      await dispatch(fetchAddresses());
+      let fetchReceiptsAction = await dispatch(fetchReciepts(pairsList));
```

Comment:
We store the order receipts inside the state, so IMO there's no need to store a copy of the state inside `fetchReceiptsAction`. We loose the single source of truth for this state when we allow copies.

Reply:
We usually only store orders/receipts for the current pair. For the rewards we need to fetch all the receipts across all pairs. So the 2 sets of data is different.
This issue will be dealt with when we relook how we deal wit State in the app.

Reply:
Sorry I think this was a misunderstanding. We do currently story all receipts across all pairs inside the state if I am not mistaken, see [reference](https://github.com/DeXter-on-Radix/website/blob/71ab1e481176eb004ec630453447c353b0448ed6/src/app/state/rewardSlice.ts#L145C7-L147C9)).

So the follow up function `fetchOrderRewards` can get  this information from state (using `thunkAPI.getState()`), it does not have to be passed in as argument to `fetchOrderRewards` IMO.

Reply:
But let's tackle in a separate PR to not block this :)

Reply:
Ahh, OK I see what you mean. But yes, I have already reworked much of this in my new state model, so lets rather review that when available and make sure we are doing it as efficiently as possible. At least we know this works for now.